scripts/sft/predict.py

```diff
@@ -0,0 +1,108 @@
+import json
+import os
+from pathlib import Path
+import time
+
+import hydra
+from datasets import load_dataset
+
+from loguru import logger
+from openai import BaseModel
+import torch
+from omegaconf import DictConfig
+from tqdm import tqdm
+from torch.utils.data import DataLoader
+from juddges.config import DatasetConfig, LLMConfig
+from juddges.settings import CONFIG_PATH
+from juddges.models.factory import get_model
+from juddges.preprocessing.text_encoder import TextEncoderForEval
+from juddges.utils.config import resolve_config
+
+DEVICE = "cuda" if torch.cuda.is_available() else "cpu"
+NUM_PROC = int(os.getenv("NUM_PROC", 1))
+
+if NUM_PROC > 1:
+    os.environ["TOKENIZERS_PARALLELISM"] = "false"
```

Comment:
is it ok?

Reply:
yes, when you parallelize by `datasets.map` and use built-in `tokenizer` parallelism simultaneously, you get warning about possible deadlocks